Annotate function args with type information

kimishpatel opened the pull request:
Summary: Without type information, default type is Tensor which may conflict at runtime.

Test Plan: CI

Differential Revision: D29990902

raziel reviewed (approved):
Thanks!

Let's just wait for @SplitInfinity to give the LGTM.

SplitInfinity commented on torch/csrc/jit/backends/backend_detail.cpp:
```diff
@@ -293,7 +293,8 @@ Module codegen_backend_module(
         std::stringstream def_ss, fwd_ss;
-        def_ss << name << "=";
+        // Annotate type of the arg
+        def_ss << name << ":" << arg.type()->annotation_str(nullptr) << "=";
```
```suggestion
        def_ss << name << ": " << arg.type()->annotation_str() << "=";
```
Not that anyone will be looking at this code but there is usually a space after the `:`

kimishpatel replied:
Yeah makes sense. Will make that change.

SplitInfinity commented on torch/csrc/jit/backends/backend_detail.cpp:
```diff
@@ -302,7 +303,10 @@ Module codegen_backend_module(
-        def_inputs.emplace_back(name);
+        std::stringstream def_ss;
+        // Annotate type of the arg
+        def_ss << name << ":" << arg.type()->annotation_str(nullptr);
```
```suggestion
        def_ss << name << ": " << arg.type()->annotation_str();
```
Same here

kimishpatel committed:
[Pytorch Backend Delegation] Annotate function args with type information (#62433)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/62433

Without type information, default type is Tensor which may conflict at runtime.

Test Plan: CI

Reviewed By: raziel

Differential Revision: D29990902

fbshipit-source-id: ca6cf81c0d6cd83984e9a31858617dfc3d0667ba
